src/util/files/resource_loader.c

```diff
@@ -0,0 +1,96 @@
+#include "resource_loader.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+//TODO: REMOVE THIS IN THE FUTURE, ONLY FOR TESTING
+//TODO: REPLACE ALL MALLOC WITH CALLOC
+#include <limits.h>
+
+
+#if defined(_WINDOWS) || defined(_WIN32) || defined(_WIN64)
+#include <direct.h>
+#define GetCurrentDirectory _getcwd
+#else
+#include <unistd.h>
+#define GetCurrentDirectory getcwd
```

Comment:
usually I would not advocate for preprocessor definitions and includes in the same block but it's kind of OK here - maybe add some comments to describe what this actually does though?